src/test/clip.t.sol

```diff
@@ -903,4 +933,36 @@ contract ClipperTest is DSTest {
         // Assert transfer of gem.
         assertEq(vat.gem(ilk, address(this)), preGemBalance + origLot);
     }
+
+    function testFail_not_enough_dai() public takeSetup {
+        Guy(che).take({
+            id:  1,
+            amt: 25 ether,
+            max: ray(5 ether),
+            who: address(che),
+            data: ""
+        });
+    }
+
+    function test_flashloan() public takeSetup {
+        assertEq(vat.dai(che), 0);
+        Guy(che).take({
+            id:  1,
+            amt: 25 ether,
+            max: ray(5 ether),
+            who: address(che),
+            data: "hey"
+        });
+        assertEq(vat.dai(che), 0);
+    }
+
+    function testFail_reentrancy() public takeSetup {
```

Comment:
You could test a second one trying to enter to `redo`.

Reply:
Added a `redo` test.